#
# Partial fix for: https://bugs.gnupg.org/gnupg/issue2144.
#
# The patch comes from the upstream. Fix for Solaris:
# http://git.gnupg.org/cgi-bin/gitweb.cgi?p=libgpg-error.git;a=commitdiff;h=34b07146bbb0863436fc9381a84314b18bdfb807
#
--- a/src/gen-posix-lock-obj.c
+++ b/src/gen-posix-lock-obj.c
@@ -43,7 +43,7 @@
 #endif
 
 /* Special requirements for certain platforms.  */
-#if defined(__solaris__) && (defined (__ILP32__) || defined(_ILP32))
+#if defined(__solaris__) && !defined (__LP64__) && !defined(_LP64)
 # define USE_DOUBLE_FOR_ALIGNMENT 1
 #else
 # define USE_DOUBLE_FOR_ALIGNMENT 0

